universal_transfer_operator/src/universal_transfer_operator/data_providers/__init__.py

```diff
@@ -5,12 +5,30 @@
 from universal_transfer_operator.constants import TransferMode
 from universal_transfer_operator.data_providers.base import DataProviders
 from universal_transfer_operator.datasets.base import Dataset
+from universal_transfer_operator.datasets.file.base import File
+from universal_transfer_operator.datasets.table import Table
 from universal_transfer_operator.utils import TransferParameters, get_class_name
 
-DATASET_CONN_ID_TO_DATAPROVIDER_MAPPING = dict.fromkeys(
-    ["s3", "aws"], "universal_transfer_operator.data_providers.filesystem.aws.s3"
-) | dict.fromkeys(
-    ["gs", "google_cloud_platform"], "universal_transfer_operator.data_providers.filesystem.google.cloud.gcs"
+DATASET_CONN_ID_TO_DATAPROVIDER_MAPPING = (
+    dict.fromkeys(
+        [("s3", File), ("aws", File)], "universal_transfer_operator.data_providers.filesystem.aws.s3"
+    )
+    | dict.fromkeys(
+        [("gs", File), ("google_cloud_platform", File)],
+        "universal_transfer_operator.data_providers.filesystem.google.cloud.gcs",
+    )
+    | dict.fromkeys(
+        [
+            ("sqlite", Table),
+        ],
+        "universal_transfer_operator.data_providers.database.sqlite",
+    )
+    | dict.fromkeys(
+        [
+            ("snowflake", Table),
+        ],
+        "universal_transfer_operator.data_providers.database.snowflake",
+    )
 )
```

Comment:
Question that doesn't necessarily need to be answered now: 

Are we going to allow users to create customer UTO extensions? If so should there be some mechanism where people can add to this map via the airflow.cfg or an env variable?